v2/apiserver/internal/system/rest/healthcheck_endpoints.go

```diff
@@ -0,0 +1,95 @@
+package rest
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/brigadecore/brigade/v2/apiserver/internal/lib/queue"
+	"github.com/brigadecore/brigade/v2/apiserver/internal/lib/restmachinery"
+	"github.com/gorilla/mux"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// HealthcheckEndpoints implements restmachinery.Endpoints to provide
+// Healthcheck-related URL --> action mappings to a restmachinery.Server.
+type HealthcheckEndpoints struct {
+	APIServerVersion string
+	DatabaseClient   *mongo.Client
+	WriterFactory    queue.WriterFactory
+}
+
+// Register is invoked by restmachinery.Server to register Healthcheck-related
+// URL --> action mappings to a restmachinery.Server.
+func (h *HealthcheckEndpoints) Register(router *mux.Router) {
+	// Get /healthz
+	router.HandleFunc(
+		"/healthz",
+		h.healthz,
+	).Methods(http.MethodGet)
+
+	// Get /ping
+	router.HandleFunc(
+		"/ping",
```

Comment:
Since the ping returns version information, do you think we should version this path? i.e. Use `/v2/ping`. This could provide two benefits:

1. Client can easily assert that the API server it's talking to is a v2 API server.
2. We get cover to change the shape of the ping response in v3 and beyond.

Reply:
Excellent foresight.  Updated.